cpp/ql/src/semmle/code/cpp/MemberFunction.qll

```diff
@@ -70,6 +70,13 @@ class MemberFunction extends Function {
       result = getADeclarationEntry() and result != getDefinition()
     )
   }
+
+  /**
+   * Gets the type of the `this` parameter associated with this member function, if any.
```

Comment:
It might be nice to spell out how this is different from `getDeclaringType()`, i.e. being a pointer and, I can see from the test, potential for `const` and `volatile` specifiers.

Reply:
I've updated the comment, but this raises an interesting question.

The `member_function_this_type` and the extractor allow, in principle, for any kind of `Type`. In practice, it's always going to be a `PointerType`. We could consider changing the return type of this predicate.

Reply:
Changing to `PointerType` sounds good to me.